src/Storages/Kafka/StorageKafka2.cpp

```diff
@@ -1322,4 +1444,11 @@ fs::path StorageKafka2::getTopicPartitionPath(const TopicPartition & topic_parti
     return fs_keeper_path / "topics" / topic_partition.topic / "partitions" / std::to_string(topic_partition.partition_id);
 }
 
+fs::path StorageKafka2::getTopicPartitionLockPath(const TopicPartition & topic_partition)
+{
+    auto topic_partition_name = topic_partition.topic + "_" + std::to_string(topic_partition.partition_id);
+    return fs_keeper_path / "topic_partition_locks" / topic_partition_name / lock_file_name;
```

Comment:
`fs_keeper_path / "topic_partition_locks" / topic_partition_name / lock_file_name;` would result in something like `/clickhouse/tables/kafka/topic_partitions_locks/topic_a_1/lock` but we need `/clickhouse/tables/kafka/topic_partitions_locks/topic_a_1.lock`.

Hint: you can use `fmt::format("{}_{}.{}", topic_partition.topic, topic_partition.partition_id, lock_file_postfix)` to create the filename.